test/performance/throughput/ThroughputPublisher.cpp

```diff
@@ -59,14 +59,21 @@ void ThroughputPublisher::DataPubListener::onPublicationMatched(
     if (info.status == MATCHED_MATCHING)
     {
         ++throughput_publisher_.data_discovery_count_;
+        std::cout << C_RED << "Pub: DATA Pub Matched " 
+            << throughput_publisher_.data_discovery_count_ << "/" << throughput_publisher_.subscribers_
+            << C_DEF << std::endl;
     }
     else
     {
         --throughput_publisher_.data_discovery_count_;
     }
 
     lock.unlock();
-    throughput_publisher_.data_discovery_cv_.notify_one();
+
+    if(throughput_publisher_.data_discovery_count_ >= static_cast<int>(throughput_publisher_.subscribers_))
```

Comment:
I think that if `data_discovery_count_ > static_cast<int>(throughput_publisher_.subscribers_)`, then we should not proceed, since we have discovered some unexpected subscriber that can affect the test results. I'd change the comparison to `==` and have and `else if` contemplating the `>` case

Reply:
The > case is an error so we shouldn't contemplate unless the decision is to abort testing.

Reply:
I think we should do that